tests/lambda_functions/va_profile/test_va_profile_integration.py

```diff
@@ -0,0 +1,247 @@
+            communication_item_id=0,
+            communication_channel_id=0,
+            allowed=False,
+            source_datetime="2022-03-07T19:37:59.320Z"
+        )
+
+        va_profile_test = VA_PROFILE_TEST.bindparams(
+            va_profile_id=0,
+            communication_item_id=0,
+            communication_channel_id=0
+        )
+
+        result = connection.execute(opt_in_out)
+        assert result.fetchone()[0], "The stored function should return True."
+
+        result = connection.execute(COUNT)
+        assert result.fetchone()[0] == 1, "The stored function should have created a new row."
+
+        result = connection.execute(va_profile_test)
+        assert not result.fetchone()[0], "The user opted out.  (allowed=False)"
+
+        opt_in_out = OPT_IN_OUT.bindparams(
+            va_profile_id=0,
+            communication_item_id=0,
+            communication_channel_id=0,
+            allowed=True,
+            source_datetime="2022-02-07T19:37:59.320Z"  # Older date
+        )
+
+        result = connection.execute(opt_in_out)
+        assert not result.fetchone()[0], "The date is older than the existing entry."
+
+        result = connection.execute(COUNT)
+        assert result.fetchone()[0] == 1, "The stored function should not have created a new row."
+
+        result = connection.execute(va_profile_test)
+        assert not result.fetchone()[0], "The user should still be opted out.  (allowed=False)"
+
+        opt_in_out = OPT_IN_OUT.bindparams(
+            va_profile_id=0,
+            communication_item_id=0,
+            communication_channel_id=0,
+            allowed=True,
+            source_datetime="2022-04-07T19:37:59.320Z"  # Newer date
+        )
+
+        result = connection.execute(opt_in_out)
+        assert result.fetchone()[0], "The date is newer than the existing entry."
+
+        result = connection.execute(COUNT)
+        assert result.fetchone()[0] == 1, "An existing entry should have been updated."
+
+        result = connection.execute(va_profile_test)
+        assert result.fetchone()[0], "The user should be opted in.  (allowed=True)"
+
+        opt_in_out = OPT_IN_OUT.bindparams(
+            va_profile_id=1,
+            communication_item_id=1,
+            communication_channel_id=1,
+            allowed=True,
+            source_datetime="2022-02-07T19:37:59.320Z"
+        )
+
+        va_profile_test = VA_PROFILE_TEST.bindparams(
+            va_profile_id=1,
+            communication_item_id=1,
+            communication_channel_id=1
+        )
+
+        result = connection.execute(opt_in_out)
+        assert result.fetchone()[0], "The stored function should have created a new row."
+
+        result = connection.execute(COUNT)
+        assert result.fetchone()[0] == 2, "The stored function should have created a new row."
+
+        result = connection.execute(va_profile_test)
+        assert result.fetchone()[0], "The user should be opted in.  (allowed=True)"
```

Comment:
I think it would be beneficial to separate the concepts for each check here. It would be easier to identify which tests we have in our code, and it would improve maintainability. IIRC there's a way to setup a group of test params so you wouldn't have to initialize the connection for each method.
```
test_add_pref()
test_remove_pref()
test_add_older_date()
etc
```